### ISSUE TITLE
Graceful shutdown hangs when I use a Chromium based browser

### ISSUE THREAD

**Issue by czy-29** [viz-rs/viz#138]
This problem of Viz is exactly the same as Axum:
https://github.com/tokio-rs/axum/issues/2611

This is due to a bug in `hyper-util`:
issue: https://github.com/hyperium/hyper/issues/3576
pr: https://github.com/hyperium/hyper-util/pull/101

Patching `hyper-util` in Cargo.toml like this also works for viz:
```
[patch.crates-io]
hyper-util = { git = "https://github.com/hyperium/hyper-util.git", rev = "a77d866ff65a68ea5d048d058108552c39ec470d" }
```

The reason for posting the same issue here in Viz is that when `hyper-util` is finally updated, I hope we can follow up with the updated version in time, so that I can remove such a temporary patch and update to the latest version.

**Comment by fundon** [comment]
Thanks for the feedback. I will fix this ASAP.

**Comment by czy-29** [comment]
hyper-util 0.1.4 is released!
https://crates.io/crates/hyper-util/0.1.4
We should update as soon as possible and finally close this issue.

**Comment by fundon** [comment]
Nice! Let's ship it.

**Comment by fundon** [comment]
Landed in v0.8.5. 
Thanks.
